File: AppointmentSchedulerAPI/layers/ServiceLayer/v1/Controllers/DTO/Response/AccountDataDTO.cs
using AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model.Types;

namespace AppointmentSchedulerAPI.layers.ServiceLayer.v1.Controllers.DTO.Response
{
    public class AccountDataDTO
    {
        public required Guid Uuid { get; set; }
        public required string Email { get; set; }
        public required string PhoneNumber { get; set; }
        public required string Username { get; set; }
        public required string Name { get; set; }
        public required RoleType Role { get; set; }
        public required AccountStatusType Status { get; set;}
        public required DateTime CreatedAt { get; set; }
    }
}

File: AppointmentSchedulerWeb/src/model/dtos/account-data.dto.ts
import { AccountStatusType } from "../../view-model/business-entities/types/account-status.types";
import { RoleType } from "../../view-model/business-entities/types/role.types";

export interface AccountDataDTO {
  uuid: string;
  email: string;
  phoneNumber: string;
  username: string;
  name: string;
  role: RoleType;
  status: AccountStatusType;
  createdAt: Date;
}
